src/base/misc/fatfs.c

```diff
@@ -1895,7 +1889,15 @@ void build_boot_blk(fatfs_t *f, unsigned char *b)
      illegal displacement of 0x00.
    */
   t_o = 0x48;
-  snprintf((char *)b + t_o, eos - t_o, "\x04\x03\x02\x01\xff" MSG_F, f->dir);
+  eos = (0x7ded-0x7c00); // 0x7ded is the maximum limit of string
+  snprintf((char *)b + t_o, eos - t_o, "\x04\x03\x02\x01\xff"
+    "\r\n"
+    "Sorry, could not load an operating system from\r\n"
+    "%s\r\n"
+    "Please check the IO.SYS file for corruption or replace with FreeDOS\r\n"
```

Comment:
> or replace with FreeDOS

Cool! :)
OK, I'll apply that pearl. :)
But next time please suggest replacing with fdpp!